### ISSUE TITLE
Performance/RegexpMatch: not enough global variables recognised

### ISSUE THREAD

**Issue by sj26** [bbatsov/rubocop#4589]
Rubocop falsely reports an offense for the RegexpMatch cop on the following code:

```ruby
if commit_info =~ /^commit \s* ([a-f0-9]+)$/xi
  info[:commit] = $1
end
```

```
code.rb:40:8: C: Use match? instead of =~ when MatchData is not used.
if commit_info =~ /^commit \s* ([a-f0-9]+)$/xi
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

The match data _is_ being used by `$1`.

To my eye this looks like the numbered global match variables (`$1`, `$2`, ...) need to be added to [`RegexpMatch#match_gvar?`](https://github.com/bbatsov/rubocop/blob/d87cea2cbaef494df11375900503784b5bc34859/lib/rubocop/cop/performance/regexp_match.rb#L199-L208).

--------

## Expected behavior

No offense reported.

## Actual behavior

Offense reported.

## Steps to reproduce the problem

Use a Regexp match and global match variables like the example above.

## RuboCop version

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by pocke** [comment]
Thanks for your report.

I can't reproduce this problem.

`test.rb`

```ruby
if commit_info =~ /^commit \s* ([a-f0-9]+)$/xi
  info[:commit] = $1
end
```

`.rubocop.yml`

```yaml
AllCops:
  TargetRubyVersion: 2.4
```

```bash
$ rubocop --only RegexpMatch --cache false
Inspecting 1 file
.

1 file inspected, no offenses detected
```


> To my eye this looks like the numbered global match variables ($1, $2, ...) need to be added to RegexpMatch#match_gvar?.

This gvar is not parsed as `gvar`, it is parsed as `nth-ref`. 
For example:

```bash
$ ruby-parse test.rb
(if
  (send
    (send nil :commit_info) :=~
    (regexp
      (str "^commit \\s* ([a-f0-9]+)$")
      (regopt :i :x)))
  (send
    (send nil :info) :[]=
    (sym :commit)
    (nth-ref 1)) nil)
```

And rubocop handle `nth-ref`. https://github.com/bbatsov/rubocop/blob/d87cea2cbaef494df11375900503784b5bc34859/lib/rubocop/cop/performance/regexp_match.rb#L106


Could you give me `ruby-parse` command output? This command is installed if rubocop is installed.

**Comment by sj26** [comment]
Apologies, that was a snippet from a larger file. When I reduced the file to only the three lines I supplied it doesn't report an offense. But if I add a little more of the context up to a class method it seems to trigger the problem:

```ruby
# frozen_string_literal: true
class MyClass
  def self.parse(commit_info)
    if commit_info =~ /^commit \s* ([a-f0-9]+)$/xi
      info[:commit] = $1
    end
  end
end
```

```
$ rubocop --only RegexpMatch --cache false app/models/scm/git.rb
Inspecting 1 file
C

Offenses:

app/models/my_class.rb:4:8: C: Use match? instead of =~ when MatchData is not used.
    if commit_info =~ /^commit \s* ([a-f0-9]+)$/xi
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

```
$ ruby-parse app/models/scm/git.rb
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
(class
  (const nil :MyClass) nil
  (defs
    (self) :parse
    (args
      (arg :commit_info))
    (if
      (send
        (lvar :commit_info) :=~
        (regexp
          (str "^commit \\s* ([a-f0-9]+)$")
          (regopt :i :x)))
      (send
        (send nil :info) :[]=
        (sym :commit)
        (nth-ref 1)) nil)))
```

**Comment by pocke** [comment]
Thanks. I can reproduce it. 👍 
I'll work on it.

**Comment by sj26** [comment]
Thank you! 🙇🏻
